odoo/addons/base/wizard/base_x509.py

```diff
@@ -0,0 +1,132 @@
+                Encoding.PEM, PublicFormat.SubjectPublicKeyInfo))
+            record.subject = cert.subject.rfc4514_string()
+            record.issuer = cert.issuer.rfc4514_string()
+            record.not_valid_before = cert.not_valid_before  # _utc.replace(tzinfo=None)
+            record.not_valid_after = cert.not_valid_after  # _utc.replace(tzinfo=None)
+
+            # replicates openssl x509 -text behavior, for easy comparison
+            record.signature = '\n'.join(textwrap.wrap(cert.signature.hex(sep=':'), 54))
+
+            try:
+                san = cert.extensions.get_extension_for_class(x509.SubjectAlternativeName)
+            except x509.ExtensionNotFound:
+                record.subject_alternative_names = ''
+            else:
+                record.subject_alternative_names = ', '.join(
+                    san.value.get_values_for_type(x509.DNSName) +
+                    san.value.get_values_for_type(x509.IPAddress))
+
+            common_names = [
+                common_name.value
+                for common_name
+                in cert.subject.get_attributes_for_oid(x509.oid.NameOID.COMMON_NAME)
+                if common_name.value
+            ]
+            if not common_names:
+                common_names = list(filter(bool, self.subject_alternative_names.split(', ')))
+            if not common_names:
+                common_names = [record.subject]
+            record.filename = clean_filename(f'{common_names[0]}.pub.pem')
+
+    def _save(self):
+        self.ensure_one()
+
+        public_key = self.env['ir.attachment'].create({
+            'name': self.filename,
+            'mimetype': 'application/x-pem-file',
+            'datas': self.public_key_pem,
+            'description': _(
+                "This public key was extracted from the following certificate.\n"
+                "\n"
+                "Subject: %(subject)s\n"
+                "Subject alternative names: %(subject_alternative_names)s\n"
+                "Issuer: %(issuer)s\n"
+                "Not valid before: %(not_valid_before)s\n"
+                "Not valid after: %(not_valid_after)s\n"
+                "Signature:\n"
+                "%(signature)s\n",
+                subject=self.subject,
+                subject_alternative_names=self.subject_alternative_names,
+                issuer=self.issuer,
+                not_valid_before=format_datetime(self.env, self.not_valid_before),
+                not_valid_after=format_datetime(self.env, self.not_valid_after),
+                signature=self.signature,
+            )
+        })
+
+        return public_key
+
+    def save(self):
+        raise NotImplementedError  # abstract method
+
+
+class BaseX509mailWizard(models.TransientModel):
+    _name = 'base.x509mail.wizard'
+    _inherit = 'base.x509.wizard'
+    _description = "x509 Certificate Import Wizard for Mail Server"
+
+    mail_server_id = fields.Many2one('ir.mail_server', required=True)
```

Comment:
This should not be required. You could still have a certificate even if you don't provide a mail server.